src/main/java/org/quiltmc/loader/impl/filesystem/GlobToRegex.java

```diff
@@ -0,0 +1,486 @@
+/*
+ * Copyright 2023 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * jimfs is Copyright (C) 2013 Google Inc, under the Apache License Version 2.0 (the same as in the header above).
+ *
+ * The following changes have been applied:
+ * - The file has been reformatted (2 space indent to tabs)
+ * - Repackaged.
+ * - "InternalCharMatcher" was partially copied into this file
+ * - "com.google.common.base.Preconditions.checkNotNull" was replaced with "java.lang.Objects.requireNonNull"
+ *
+ * You may view the original, including its license header, here:
+ * https://github.com/google/jimfs/blob/64b0b187221a529be531271d9e27627ad1954572/jimfs/src/main/java/com/google/common/jimfs/GlobToRegex.java
+ * 
+ * In addition the file "InternalCharMatcher" was partially copied into this file, under the same license, from here:
+ * https://github.com/google/jimfs/blob/4739aee7e8e6ee9c66801541921f42f9af93387d/jimfs/src/main/java/com/google/common/jimfs/InternalCharMatcher.java
+ */
```

Comment:
Yeah, this is fine. It's technically way overboard for what the license requires, but it's (1) best not to poke the bear and (2) useful for when someone else comes down the line in 7 years (which isn't super unrealistic, considering parts of Loader still present today were written in 2016) and has to figure out wtf this is.